### ISSUE TITLE
True color issue when launching tmux from default_prog

### ISSUE THREAD

**Issue by alexhausen** [wez/wezterm#2852]
### What Operating System(s) are you seeing this problem on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

Ubuntu 22.04.5
Gnome 3.36.8
X11

### WezTerm version

wezterm 20221119-145034-49b9839f

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Tmux doesn't display true colors when launched from `default_prog`.

### To Reproduce

Add the following to `wezterm.lua` config file:
`default_prog = { '/usr/bin/tmux' },`

Then verify true color output.
I've used the script from https://jdhao.github.io/2018/10/19/tmux_nvim_true_color/#verifying

It displays the 256 colors output.

### Configuration

`default_prog = { '/usr/bin/tmux' },`

### Expected Behavior

It should have displayed the true color output of the script from https://jdhao.github.io/2018/10/19/tmux_nvim_true_color/#verifying

### Anything else?

When the `default_prog` option is not used and I start tmux manually from the shell prompt it displays truecolor correctly.

Side note: when `default_prog` is set to `/usr/bin/dash`  it also works.
So it may not be an issue with Wezterm.

**Comment by wez** [comment]
When using `default_prog`, wezterm doesn't use your shell to launch the program.
If you rely on your shell to setup your environment for tmux, then things may be missing from your environment.
Two alternative approaches to consider:

1. Explicitly launch tmux via your shell:

```lua
default_prog = {"bash", "-c", "exec /usr/bin/tmux"}
```

2. Explicitly set the environment:

```lua
default_prog = {"/usr/bin/tmux"},
set_environment_variables={
   -- set whatever environment is needed to make tmux use true color
   -- Educated guess is terminfo:
   TERMINFO_DIRS = "something",
}
```

See also: https://github.com/tmux/tmux/wiki/FAQ#how-do-i-use-rgb-colour

**Comment by alexhausen** [comment]
Thank you so much for such quick response.

I've tried both suggestions without success.

However, I've found the solution after executing `tmux info | grep Tc` on both true-color enabled and not enabled terminals.

Terminal with true-color:
```
$ tmux info | grep Tc
206: Tc: (flag) true
```

Terminal without true-color:
```
$ tmux info | grep Tc
206: Tc: [missing]
206: Tc: (flag) true
```

For some reason the `TERM` variable set on my `.bashrc` (`tmux-256color`) was different from the default value when running from `default_prog` (`xterm-256color`).

Removing the `TERM` from `.bashrc` fixed the issue after a new login.
